Framework/Algorithms/src/PolarisedSANS/HeliumAnalyserEfficiency.cpp

```diff
@@ -0,0 +1,356 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "MantidAlgorithms/PolarisedSANS/HeliumAnalyserEfficiency.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/HistogramValidator.h"
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidAPI/WorkspaceUnitValidator.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidKernel/ListValidator.h"
+
+#include <algorithm>
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/join.hpp>
+#include <boost/math/distributions/students_t.hpp>
+#include <vector>
+
+namespace Mantid::Algorithms {
+// Register the algorithm into the algorithm factory
+DECLARE_ALGORITHM(HeliumAnalyserEfficiency)
+
+using namespace Kernel;
+using namespace API;
+
+namespace SpinConfigurations {
+static const std::string UpUp = "11";
+static const std::string UpDown = "10";
+static const std::string DownUp = "01";
+static const std::string DownDown = "00";
+} // namespace SpinConfigurations
+
+namespace PropertyNames {
+static const std::string INPUT_WORKSPACE = "InputWorkspace";
+static const std::string OUTPUT_T_WORKSPACE = "OutputTransmissionWorkspace";
+static const std::string P_HE = "HeliumPolarisation";
+static const std::string OUTPUT_T_PARA_WORKSPACE = "OutputTransmissionParaWorkspace";
+static const std::string OUTPUT_T_ANTI_WORKSPACE = "OutputTransmissionAntiWorkspace";
+static const std::string SPIN_CONFIGURATIONS = "SpinConfigurations";
+static const std::string T_E = "TransmissionEmptyCell";
+static const std::string PXD = "GasPressureTimesCellLength";
+static const std::string COVARIANCE = "Covariance";
+static const std::string START_LAMBDA = "StartLambda";
+static const std::string END_LAMBDA = "EndLambda";
+static const std::string STOP_ON_FIT_ERROR = "StopOnFitError";
+} // namespace PropertyNames
+
+/// Empty default constructor algorithm() calls the constructor in the base class
+HeliumAnalyserEfficiency::HeliumAnalyserEfficiency() : Algorithm() {}
+
+void HeliumAnalyserEfficiency::init() {
+  // Declare required input parameters for algorithm and do some validation here
+  auto val = std::make_shared<CompositeValidator>();
+  val->add<WorkspaceUnitValidator>("Wavelength");
+  val->add<HistogramValidator>();
+  declareProperty(std::make_unique<WorkspaceProperty<>>(PropertyNames::INPUT_WORKSPACE, "", Direction::Input, val));
+  declareProperty(std::make_unique<WorkspaceProperty<>>(PropertyNames::OUTPUT_T_WORKSPACE, "T", Direction::Output));
+  declareProperty(std::make_unique<WorkspaceProperty<>>(PropertyNames::P_HE, "p_He", Direction::Output));
```

Comment:
I'm not sure yet which are going to be the most used outputs, because e.g. in the polarizer efficiency algorithm we're going to be using the `T_para` and `T_anti` workspaces.